compiler/modules/front/codeGen.bal

```diff
@@ -1311,7 +1320,7 @@ function codeGenTypeTest(CodeGenContext cx, bir:BasicBlock bb, Environment env,
     return { result, block: nextBlock, narrowing };   
 }
 
-function codeGenCheckingExpr(CodeGenContext cx, bir:BasicBlock bb, Environment env, s:CheckingKeyword checkingKeyword, s:Expr expr) returns CodeGenError|ExprEffect {
+function codeGenCheckingExpr(CodeGenContext cx, bir:BasicBlock bb, Environment env, s:CheckingKeyword checkingKeyword, s:Expr expr, boolean isStmt) returns CodeGenError|RegExprEffect|StmtEffect {
```

Comment:
I think we can improve this. I don't like have something called codeGenCheckingExpr that handles both expressions and statements. I suggest we have both codeGenCheckingStmt and a codeGenCheckingExpr that both codeGen the subExpr and then feed into a common function. The codeGenCheckingStmt will check whether the operand is a subtype of t:ERROR and if so handle that case, before feeding into the common function.